djstripe/contrib/rest_framework/serializers.py

```diff
@@ -8,12 +8,197 @@
 """
 
 from rest_framework import serializers
-from rest_framework.serializers import ModelSerializer
+from rest_framework.exceptions import MethodNotAllowed
+from rest_framework.serializers import ModelSerializer, ValidationError
 
-from djstripe.models import Subscription
+from djstripe.enums import SubscriptionStatus
+from djstripe.models import Plan, Subscription
+from djstripe.settings import CANCELLATION_AT_PERIOD_END
 
+from .mixins import AutoCustomerModelSerializerMixin
 
-class SubscriptionSerializer(ModelSerializer):
+
+class SubscriptionSerializer(AutoCustomerModelSerializerMixin, ModelSerializer):
+    """A base serializer used for the Subscription model."""
+
+    class Meta:
+        model = Subscription
+        exclude = ["default_tax_rates"]
+
+    # StripeModel fields
+    id = serializers.CharField(required=False)
```

Comment:
As per comment, I think most of these fields should be read-only.

maybe `Meta.read_only_fields` should be set?